### PR TITLE
Network manager

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/39494

- Move wallet network related code from `brave_wallet_utils.cc` to a `network_manager.cc` to be managed by a single instance of `NetworkManager`
- Remove outdated prefs migrations `kBraveWalletTransactionsChainIdMigrated` and `kBraveWalletSolanaTransactionsV0SupportMigrated`

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by onyb — APPROVED** [review]

**Review by iefremov — APPROVED** [review]
browser/net lgtm

**Review by StephenHeaps — APPROVED** [review]
iOS++

File: browser/ui/webui/settings/brave_wallet_handler.cc [hunk 1]
```diff
@@ -356,9 +351,7 @@ void BraveWalletHandler::AddHiddenNetwork(const base::Value::List& args) {
     return;
   }
 
-  PrefService* prefs = GetPrefs();
-  AllowJavascript();
```
Comment (yrliou) [inline browser/ui/webui/settings/brave_wallet_handler.cc:360]: why is this line removed?

File: components/brave_wallet/browser/network_manager_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,1050 @@
...
+//   std::vector<base::Value::Dict> values;
+//   values.push_back(value.Clone());
+//   UpdateCustomNetworks(prefs, values, coin);
+// }
+
+// std::vector<brave_wallet::mojom::NetworkInfoPtr> GetAllEthCustomChains(
+//     NetworkManager* network_manager) {
+//   return network_manager->GetAllCustomChains(
+//       brave_wallet::mojom::CoinType::ETH);
+// }
```
Comment (yrliou) [inline components/brave_wallet/browser/network_manager_unittest.cc:78]: Why leave these?

File: components/brave_wallet/browser/network_manager_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,1050 @@
...
+    network_manager()->AddCustomNetwork(chain2);
+
+    // std::vector<base::Value::Dict> values;
+
+    // mojom::NetworkInfo chain1 = GetTestNetworkInfo1("chain1", coin);
+    // values.push_back(NetworkInfoToValue(chain1));
+
+    // mojom::NetworkInfo chain2 = GetTestNetworkInfo2("chain2", coin);
+    // values.push_back(NetworkInfoToValue(chain2));
+    // UpdateCustomNetworks(prefs(), std::move(values), coin);
```
Comment (yrliou) [inline components/brave_wallet/browser/network_manager_unittest.cc:114]: what are these?

File: components/brave_wallet/browser/network_manager.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  void RemoveCustomNetwork(const std::string& chain_id, mojom::CoinType coin);
+
+  std::vector<std::string> GetHiddenNetworks(mojom::CoinType coin);
+  void AddHiddenNetwork(mojom::CoinType coin, const std::string& chain_id);
+  void RemoveHiddenNetwork(mojom::CoinType coin, const std::string& chain_id);
+
+  // Get/Set the current chain ID for coin from kBraveWalletSelectedNetworks
+  // pref
+  // when origin is not presented. If origin is presented,
```
Comment (yrliou) [inline components/brave_wallet/browser/network_manager.h:67]: indent here is weird.

Comment (supermassive) [reply components/brave_wallet/browser/network_manager.h]: fixed by https://github.com/brave/brave-core/pull/24365/commits/90f3a806f44fb0020b62971c6a305622a0c6a362

File: components/brave_wallet/browser/network_manager_unittest.cc (hunk 3 above)
Comment (supermassive) [reply components/brave_wallet/browser/network_manager_unittest.cc]: cleaned this: https://github.com/brave/brave-core/pull/24365/commits/90f3a806f44fb0020b62971c6a305622a0c6a362

File: components/brave_wallet/browser/network_manager_unittest.cc (hunk 2 above)
Comment (supermassive) [reply components/brave_wallet/browser/network_manager_unittest.cc]: cleaned this: https://github.com/brave/brave-core/pull/24365/commits/90f3a806f44fb0020b62971c6a305622a0c6a362

File: browser/ui/webui/settings/brave_wallet_handler.cc (hunk 1 above)
Comment (supermassive) [reply browser/ui/webui/settings/brave_wallet_handler.cc:360]: Unintentional. Fixed: https://github.com/brave/brave-core/pull/24365/commits/90f3a806f44fb0020b62971c6a305622a0c6a362

**Review by yrliou — APPROVED** [review]
